packages/babel-helpers/src/helpers.js

```diff
@@ -434,16 +434,23 @@ helpers.construct = () => template.program.ast`
   import setPrototypeOf from "setPrototypeOf";
 
   export default function _construct(Parent, args, Class) {
-    if (typeof Reflect !== "undefined" && Reflect.construct) {
-      _construct = Reflect.construct;
+    if (
+      typeof Reflect !== "undefined" && Reflect.construct
+    ) {
+      _construct = function _construct(Parent, args, Class) {
+        // This wrapper is needed because Reflect.construct can't be properly
+        // polyfilled, thus core-js doesn't set the correct __proto__.
+        if (Class === undefined) return Reflect.construct(Parent, args);
+        var result = Reflect.construct.apply(null, arguments);
+        if (!(result instanceof Class)) result = setPrototypeOf(result, Class.prototype);
+        return result;
+      };
     } else {
       _construct = function _construct(Parent, args, Class) {
-        var a = [null];
+        var Constructor, a = [null];
         a.push.apply(a, args);
-        var Constructor = Parent.bind.apply(Parent, a);
-        var instance = new Constructor();
-        if (Class) setPrototypeOf(instance, Class.prototype);
-        return instance;
+        Constructor = Parent.bind.apply(Parent, a);
+        return setPrototypeOf(new Constructor, Class.prototype);
```

Comment:
Where were these reverted? This supports `null` as a super class (other transforms use this helper).